logstash-core/lib/logstash/inputs/metrics.rb

```diff
@@ -35,19 +36,22 @@ def stop
     def update(snapshot)
       @logger.debug("Metrics input: received a new snapshot", :created_at => snapshot.created_at, :snapshot => snapshot, :event => snapshot.metric_store.to_event) if @logger.debug?
 
-      # TODO: (ph)
-      # - Obviously the format here is wrong and we need to
-      # transform it from Snapshot to an event
-      # - There is another problem, if the queue is full this could block the snapshot thread.
-      # There is a few possible solution for this problem:
-      #   - We can use a future
-      #   - We can use a synchronization mechanism between the called thread (update method)
-      #   and the plugin thread (run method)
-      #   - How we handle back pressure here?
-      #   - one snashot should be only one event ?
+      # The back pressure is handled in the collector's
+      # scheduled task (running into his own thread) if something append to one of the listener it will
+      # will timeout. In a sane pipeline, with a low traffic of events it shouldn't be a problems.
       snapshot.metric_store.each do |metric|
-        @queue << LogStash::Event.new({ "@timestamp" => snapshot.created_at }.merge(metric.to_hash))
+        @queue << LogStash::Event.new({ "@timestamp" => snapshot.created_at }.merge(convert_to_hash(metric)))
```

Comment:
It could but If I remember correctly LogStash::Json.dump(hash) will call to_hash on the object before calling `to_json`, I will double check it.